test/controllers/FileChecksResultsControllerSpec.scala

```diff
@@ -0,0 +1,80 @@
+package controllers
+
+import java.util.UUID
+
+import com.github.tomakehurst.wiremock.WireMockServer
+import com.github.tomakehurst.wiremock.client.WireMock.{okJson, post, urlEqualTo}
+import configuration.GraphQLConfiguration
+import play.api.test.FakeRequest
+import play.api.test.Helpers._
+import services.ConsignmentService
+import util.FrontEndTestHelper
+
+import scala.concurrent.ExecutionContext
+
+class FileChecksResultsControllerSpec extends FrontEndTestHelper {
+  implicit val ec: ExecutionContext = ExecutionContext.global
+
+  val consignmentId: UUID = UUID.fromString("0a3f617c-04e8-41c2-9f24-99622a779528")
+  val wiremockServer = new WireMockServer(9006)
+
+  override def beforeEach(): Unit = {
+    wiremockServer.start()
+  }
+
+  override def afterEach(): Unit = {
+    wiremockServer.resetAll()
+    wiremockServer.stop()
+  }
+
+  "FileChecksResultsController GET" should {
+
+    "render the fileChecksResults page with the confirmation box" in {
+      val graphQLConfiguration = new GraphQLConfiguration(app.configuration)
+      val consignmentService = new ConsignmentService(graphQLConfiguration)
+
+      val exampleApiReturn = "{\"data\":{\"getConsignment\":{\"totalFiles\":12,\"parentFolder\": \"this-is-a-test\"}}}"
+
+      wiremockServer.stubFor(post(urlEqualTo("/graphql"))
+        .willReturn(okJson(exampleApiReturn)))
+
+      val fileCheckResultsController = new FileChecksResultsController(
+        getAuthorisedSecurityComponents,
+        getValidKeycloakConfiguration,
+        new GraphQLConfiguration(app.configuration),
+        consignmentService,
+        frontEndInfoConfiguration
+      )
+
+      val recordCheckResultsPage = fileCheckResultsController.fileCheckResultsPage(consignmentId).apply(
+        FakeRequest(GET, s"consignment/$consignmentId/records-results")
+      )
+      val resultsPageAsString = contentAsString(recordCheckResultsPage)
+
+      status(recordCheckResultsPage) mustBe 200
+      contentType(recordCheckResultsPage) mustBe Some("text/html")
+      resultsPageAsString must include("fileChecksResults.header")
+      resultsPageAsString must include("fileChecksResults.title")
+      resultsPageAsString must include("govuk-panel--confirmation")
+      resultsPageAsString must include("fileChecksResults.description")
+      resultsPageAsString must include("fileChecksResults.dashboard")
+      resultsPageAsString must include("fileChecksResults.continueLink")
+    }
+
+    "return a redirect to the auth server with an unauthenticated user" in {
```

Comment:
It would be good to add a test for an authenticated user that does not own the consignment. I don't know if there are examples of this kind of thing already.

Reply:
I'll give it a go :+1: